tools/boilerplate.py

```diff
@@ -137,155 +140,108 @@ def boilerplate_gen():
-            # be retired soon.
-            if not varargs and fmt is PLOT_TEMPLATE and has_data:
-                args.pop()
-                defaults = defaults[:-1]
-
-            if varargs is not None:
-                call.append('*' + varargs)
-            if varkw is not None:
-                call.append('**' + varkw)
-            call = ', '.join(call)
-
-            text_wrapper.width = 80 - 19 - len(func)
-            join_with = '\n' + ' ' * (18 + len(func))
-            call = join_with.join(text_wrapper.wrap(call))
-
-            if not varargs and fmt is PLOT_TEMPLATE and has_data:
-                args.append('data')
-                defaults = defaults + (None,)
-
-            # Now we can build the argspec for defining the wrapper
-            argspec = inspect.formatargspec(args, varargs, varkw, defaults,
-                                            formatvalue=format_value)
-            argspec = argspec[1:-1]  # remove parens
-
-            text_wrapper.width = 80 - 5 - len(func)
-            join_with = '\n' + ' ' * (5 + len(func))
-            argspec = join_with.join(text_wrapper.wrap(argspec))
-
-            # A gensym-like facility in case some function takes an
-            # argument named washold, ax, or ret
-            washold, ret, ax = 'washold', 'ret', 'ax'
-            bad = {*args, varargs, varkw}
-            while washold in bad or ret in bad or ax in bad:
-                washold = 'washold' + str(random.randrange(10 ** 12))
-                ret = 'ret' + str(random.randrange(10 ** 12))
-                ax = 'ax' + str(random.randrange(10 ** 12))
-
-            # Since we can't avoid using some function names,
-            # bail out if they are used as argument names
-            for reserved in ('gca', 'gci'):
-                if reserved in bad:
-                    msg = 'Axes method %s has kwarg named %s' % (func, reserved)
-                    raise ValueError(msg)
-
-            yield fmt % locals()
+                self._repr = repr(value)
+
+        def __repr__(self):
+            return self._repr
+
+    text_wrapper = textwrap.TextWrapper(
+        break_long_words=False, width=70,
+        initial_indent=' ' * 8, subsequent_indent=' ' * 8)
+
+    for func in _commands:
+        # For some commands, an additional line is needed to set the color map.
+        if func in cmappable:
+            fmt = CMAPPABLE_TEMPLATE
+            mappable = '    ' + cmappable[func]
+        else:
+            fmt = NON_CMAPPABLE_TEMPLATE
+
+        # Get signature of wrapped function.
+        sig = inspect.signature(getattr(Axes, func))
+
+        # Replace self argument.
+        params = list(sig.parameters.values())[1:]
+
+        sig = str(sig.replace(parameters=[
+            param.replace(default=value_formatter(param.default))
+            if param.default is not param.empty else param
+            for param in params]))
+        if len('def ') + len(func) + len(sig) >= 80:
```

Comment:
len('def ' + func + sig)

Reply:
I think you can merge after this (effectively there's been two review approvals) (or even without fixing it if you don't want to).

Reply:
I thought I won't construct a new string. But you're right. This is unneccesary micro-optimization.

Reply:
Ok, will merge after CI pass, or do it yourself if you are around more early than me.